src/php/DataSift/Storyplayer/HostLib/VagrantVm.php

```diff
@@ -142,17 +142,18 @@ public function createHost($vmDetails, $provisioningVars = array())
 
 		// yes it did!!
 
-		// Set automatic rsync mode
-		// This command watches all local directories of any rsync synced folders
-		// and automatically initiates an rsync transfer when changes are detected.
-		// the "--poll" option is required for some filesystems that don't support events.
-		$command = 'vagrant rsync-auto --poll';
-		$result = $log->addStep('set automatic rsync mode for Vagrant', function() use($envDetails, $command) {
-			return $this->runCommandAgainstHostManager($envDetails, $command);
-		});
-
-		if ($result->returnCode != 0) {
-			$st->usingLog()->writeToLog('WARNING: unable to set the rsync-auto mode, you need at least Vagrant 1.6.4');
+		if (getenv('VAGRANT_AUTO_RSYNC') === 'true') {
+			// Set automatic rsync mode (requires Vagrant >= 1.7.0)
+			// This command watches all local directories of any rsync synced folders
+			// and automatically initiates an rsync transfer when changes are detected.
+			// the "--poll" option is required for some filesystems that don't support events.
+			$command = 'vagrant rsync-auto --poll';
```

Comment:
`poll` is probably one of the reasons that this performs slowly. inotify would work much faster and require less resources